networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/provider_nsx_policy.py

```diff
@@ -217,16 +217,7 @@ def infra(self, target_obj: dict, child_objs: List[dict]) -> dict:
                     "id": target_obj.get("id"),
                     "target_type": target_obj.get("resource_type"),
                     "children": [
-                        {
-                            o.get("resource_type"): {
-                                "path": o.get("path"),
-                                "parent_path": o.get("parent_path"),
-                                "id": o.get("id"),
-                                "resource_type": o.get("resource_type")
-                            },
-                            "resource_type": "Child{}".format(o.get("resource_type")),
-                            "marked_for_delete": o.get("marked_for_delete")
-                        } for o in child_objs
+                        { o } for o in child_objs
```

Comment:
This simplification improvement does not work for the delete segment port scenario. Also, this is the reason why one of the functional tests failed. Suggest reviewing it to the old structure and adding the additional fields you need (if any). Also, it will be more readable.